platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -510,81 +511,106 @@ public String getFactorySnippet(@ApiParam(value = "Factory ID", required = true)
+
+        return Response.ok(factory, MediaType.APPLICATION_JSON)
+                       .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=factory.json")
+                       .build();
+    }
+
+    private static WorkspaceConfigDto asDto(UsersWorkspaceImpl workspace, String projectPath) throws ConflictException {
+        final List<CommandDto> commands = workspace.getCommands()
+                                                   .stream()
+                                                   .map(DtoConverter::asDto)
+                                                   .collect(toList());
+
+        // Filter out projects by path and source storage presence.
+        Predicate<ProjectConfigImpl> predicate = new Predicate<ProjectConfigImpl>() {
+            @Override
+            public boolean test(ProjectConfigImpl projectConfig) {
+
+                if (projectPath != null) {
+                    if (!projectConfig.getPath().equals(projectPath)) {
+                        return false;
+                    }
+                }
+
+                return projectConfig.getSource() != null &&
+                       projectConfig.getSource().getType() != null &&
+                       !projectConfig.getSource().getType().isEmpty() &&
+                       projectConfig.getSource().getLocation() != null &&
+                       !projectConfig.getSource().getLocation().isEmpty();
             }
-        } catch (IOException e) {
-            throw new ServerException(e.getLocalizedMessage());
+        };
+
+        final List<ProjectConfigDto> projects = workspace.getProjects()
+                                                         .stream()
+                                                         .filter(predicate)
+                                                         .map(DtoConverter::asDto)
+                                                         .collect(toList());
+
+        if (projects.isEmpty()) {
+            throw new ConflictException(
+                    "Unable to create factory from this workspace, because it does not contains projects with source storage set and/or specified path");
         }
-        return Response.ok(dtoFactory.createDto(FactoryV4_0.class)
-                                     .withV("4.0"), MediaType.APPLICATION_JSON)
-                       .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=" + path + ".json")
-                       .build();
+
+        final Map<String, EnvironmentDto> environments = workspace.getEnvironments()
+                                                                  .values()
+                                                                  .stream()
+                                                                  .collect(toMap(EnvironmentStateImpl::getName, FactoryService::asDto));
+
+        return DtoFactory.newDto(WorkspaceConfigDto.class)
+                                              .withName(workspace.getName())
```

Comment:
It is not formatted, `.withName` and other chain methods should be aligned with `.newDto` see [doc](https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#for-dot--separator-break-always-comes-before-the-symbol)